py-polars/polars/internals/series/datetime.py

```diff
@@ -911,7 +917,7 @@ def cast_time_unit(self, tu: TimeUnit) -> pli.Series:
 
         """
 
-    def with_time_zone(self, tz: str) -> pli.Series:
+    def convert_time_zone(self, tz: str) -> pli.Series:
```

Comment:
If we're deprecating stuff anyway, can we take a moment to consider the parameter name here?

This parameter name is inconsistent with for example `date_range`, where it's called `time_zone`. I like the latter better, since it's more descriptive, and the brevity isn't worth much as you do not need to use a keyword argument here anyway.

Not sure how this works with the redirect, but I think we're safe if we add the `@deprecated_alias` decorator here?

Reply:
yup, good call! 

OK to keep it to a separate PR to keep the scope down? The diff is already looking quite large

Reply:
Separate PR is fine, but ideally it would be part of the same release, so people don't have to update their code twice.

Reply:
agreed - and yes, I'll make sure to do a follow-up quite quickly